assets/js/projectComponents/pages/components/MarkdownEditor.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+    <v-row>
+        <v-col cols="6">
+            <textarea v-model="editorContent" @input="onInput" rows="20"></textarea>
+        </v-col>
+        <v-col cols="6">
+            <div v-html="compiledMarkdown"></div>
+        </v-col>
+    </v-row>
+</template>
+
+<script>
+import marked from 'marked'
+import DOMPurify from 'dompurify'
+import { debounce } from 'lodash'
+
+export default {
+    name: "MarkdownEditor",
+    props: {
+        content: String
+    },
+    mounted() {
+        this.editorContent = this.content
+        this.compileMarkdown()
+    },
+    data() {
+        return {
+            editorContent: null,
+            compiledMarkdown: null
+        }
+    },
+    methods: {
+        compileMarkdown() {
+            if (this.editorContent) {
+                this.compiledMarkdown = DOMPurify.sanitize(marked(this.editorContent))
+            }
+        },
+        onInput: debounce(function() {
+            this.compileMarkdown()
+            this.$emit('input', { content: this.editorContent, compiledMarkdown: this.compiledMarkdown })
```

Comment:
Is the compiledMarkedown used anywhere? If not let's remove this so we don't need the compileMarkdown and use the viewer component. To unify the compilation you can introduce a compileMarkdown.js module so it can be used where and when we need.